torch/_ops.py

```diff
@@ -122,6 +122,14 @@ def inner(fn):
                     f"Trying to override a python impl for {k} on operator {self.name()}"
                 )
             self.py_kernels[k] = fn
+            if k == torch._C.DispatchKey.CompositeImplicitAutograd and torch._C.DispatchKey.Functionalize not in self.py_kernels:
+                # Functionalization codegen is sneaky,
+                # and it registers C++ CompositeImplicit kernels *directly* to the Functionalize key.
+                # If we have a CompositeImplicit decomp registered from python, we want functionalize
+                # to use it, instead of the C++ decomp. We can't though, because Functionalize
+                # isn't part of the CompositeImplicitAutograd alias set.
+                # (open quesetion: will we eventually need to do this for functorch transform keys too?)
+                self.py_kernels[torch._C.DispatchKey.Functionalize] = fn
```

Comment:
A less hacky version of this would be to modify our py_impl CompositeImplicitAutograd sites to call some higher level function which takes care of doing both registrations. This would be good because the hack as written violates the invariant that one py_impl does one registration.